feat(impl):[TRI-1440] known knows (#413)
catenax-ng/tx-item-relationship-service: CHANGELOG.md

Reviewer: Maybe we want to add that we are currently not really managing our resources (e.g. max threads)

- The client code for accessing the Digital Twin Registry (central and decentral) is now available as a spring boot maven library. See the README in the irs-registry-client module for more information.

### Known knowns
- [TRI-1460] ESS Notifications endpoints are not working in the decentral Digital Twin Registry scenario because endpoints does not provide bpn as a parameter. 
- [TRI-1096] No limiting of requests in parallel - IRS allows sending API requests unlimited
- [TRI-1100] Potential denial-of-service (DoS) attack - IRS allows to enter a large number of characters, which are reflected in the response of the server
- [TRI-1098] Software related information disclosure - IRS returns redundant information about the type and version of used software
- [TRI-793] Misconfigured Access-Control-Allow- Origin Header - by intercepting network traffic it could be possible to read and modify any messages that are exchanged with server
- [TRI-1095] HTTP security headers configuration could be improved and allow for additional protection against some web application attacks

> - Synchronous communication with external services without circuit breaker pattern - potentially could affect IRS resilience when other services becomes non-responsive

Reviewer 2: ```suggestion
- [TRI-1441] Synchronous communication with shared C-X services without circuit breaker pattern - potentially could affect IRS resilience when other services becomes non-responsive.
```

> - Cascading effects of failure when Digital Twin Registry becomes non-responsive - potentially bulkhead pattern could improve IRS resilience

Reviewer 2: ```suggestion
- [TRI-1441] Cascading effects of failure when Digital Twin Registry becomes non-responsive - potentially bulkhead pattern could improve IRS resilience
```

> - Retry mechanism used inside IRS could potentially affect IRS resilience - DDOS other services on which IRS is dependent, exhaustion of resources and available threads, etc.

Reviewer 2: ```suggestion
- [TRI-1477] Retry mechanism used inside IRS could potentially affect IRS resilience - DDOS other services on which IRS is dependent, exhaustion of resources and available threads, etc.
```

> - Lack of resources management - max threads, heap limitation, etc.

Reviewer 2: ```suggestion
- [TRI-1478] Lack of resources management - thread pooling, heap limitation etc.
```

> - IRS was not scaled to multiple instances, and was not tested in such environment

Reviewer 2: ```suggestion
- [TRI-1024] IRS does not support scale out on multiple instances
```

- The client code for accessing the Digital Twin Registry (central and decentral) is now available as a spring boot maven library. See the README in the irs-registry-client module for more information.

### Known knowns
- [TRI-1346] Digital Twin Registry(DTR) - performance issues - after subsequent requests DTR starts responding 500 http status code
- [TRI-1460] ESS Notifications endpoints are not working in the decentral Digital Twin Registry scenario because endpoints does not provide bpn as a parameter. 

> - [TRI-1455] Random 500 http status code returned by EDC during submodel retrievals. (EDC version: 0.4.1)

Reviewer 3: Not the case anymore with the latest IRS and EDC version

> - Lack of rate limiting of requests - IRS allows sending API requests in a massive, automated manner

Reviewer 2: ```suggestion
- [TRI-1441] - No limiting of requests in paralle - IRS allows sending API requests unlimited
```

> - High EDC resource demands - each EDC provider postgres uses > 2 CPU-Cores. The database storage gets filled after few days, until then, resource consumption increases. (EDC version: 0.4.1)

Reviewer 3: Since EDC 0.4.1 this is not an issue anymore

Reviewer 3: I already closed it

> - Potential denial-of-service (DoS) attack - IRS allows to enter a large number of characters, which are reflected in the response of the server

Reviewer 2: Could you please more clear which parameter which ticket please?

> - Software related information disclosure - IRS returns redundant information about the type and version of used software

Reviewer 2: What is this exactly? how to reproduce please add a ticket.
